datasources/src/commonMain/kotlin/com/mirego/trikot/datasources/extensions/DataStateExtensions.kt

```diff
@@ -14,3 +15,18 @@ fun <T, E : Throwable> DataState<T, E>.error(): E? =
         is DataState.Error -> this.error
         else -> null
     }
+
+fun <T, R, E : Throwable> DataState<T, E>.mapData(block: MapProcessorBlock<T, R>): DataState<R, E> =
+    when (this) {
+        is DataState.Pending -> DataState.Pending(value?.let(block))
+        is DataState.Data -> DataState.Data(block(value))
+        is DataState.Error -> DataState.Error(error, value?.let(block))
+    }
+
+fun <T, E : Throwable> DataState<T, E>.extractValue(): T? {
```

Comment:
I don't think the name `extractValue` really represent . The best name would have been `value` but I think its not possible cause of the sub implementation.

I don't have anything clever to suggest except `anyStateValue`. Better ideas ?

Reply:
Actually since it's a function, value() would probably work. (would not clash with the val)

Reply:
Or we could go with a property extension instead
`val <T, E : Throwable> DataState<T, E>.value: T? get= ...`